src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -511,20 +511,6 @@ bool CascadiaSettings::_AppendDynamicProfilesToUserSettings()
 
     for (const auto& profile : _profiles)
     {
-        if (profile.Guid() == winrt::guid{})
-        {
-            // If the profile doesn't have a guid, it's a name-only profile.
-            // During validation, we'll generate a GUID for the profile, but
-            // validation occurs after this. We should ignore these types of
-            // profiles.
-            // If a dynamic profile was generated _without_ a GUID, we also
-            // don't want it serialized here. The first check in
-            // Profile::ShouldBeLayered checks that the profile has a guid. For a
-            // dynamic profile without a GUID, that'll _never_ be true, so it
-            // would be impossible to be layered.
-            continue;
-        }
-

```

Comment:
this might change the behavior of guid-less dynamic profiles. is that acceptable? we don't think any exist today.

Reply:
I think it is? We get a collision when two profiles have...
- the same guid
- the same name & source --> same deterministically auto-generated guid

This seems relevant to proto-extensions. Authors need to claim a guid. Or (on our end) we need to dig deeper into what the "source" is for a proto-extension (not just "local"). Both seem reasonable to me.